pxf/pxf-ignite/src/main/java/org/apache/hawq/pxf/plugins/ignite/IgniteResolver.java

```diff
@@ -0,0 +1,173 @@
+                fields.add(oneField);
+                continue;
+            }
+            switch (DataType.get(oneField.type)) {
+                case INTEGER:
+                    value = result.get(i).getAsInt();
+                    break;
+                case FLOAT8:
+                    value = result.get(i).getAsDouble();
+                    break;
+                case REAL:
+                    value = result.get(i).getAsFloat();
+                    break;
+                case BIGINT:
+                    value = result.get(i).getAsLong();
+                    break;
+                case SMALLINT:
+                    value = result.get(i).getAsShort();
+                    break;
+                case BOOLEAN:
+                    value = result.get(i).getAsBoolean();
+                    break;
+                case VARCHAR:
+                case BPCHAR:
+                case TEXT:
+                case NUMERIC:
+                    value = result.get(i).getAsString();
+                    break;
+                case BYTEA:
+                    value = Base64.decodeBase64(result.get(i).getAsString());
+                    break;
+                case TIMESTAMP:
+                    value = new SimpleDateFormat("MMM d, yyyy hh:mm:ss a").parse(result.get(i).getAsString());
+                    break;
+                case DATE:
+                    value = new SimpleDateFormat("yyyy-MM-dd").parse(result.get(i).getAsString());
+                    break;
+                default:
+                    throw new UnsupportedOperationException("Field type not supported: " + DataType.get(oneField.type).toString()
+                            + ", Column : " + columns.get(i).columnName());
+            }
+
+            oneField.val = value;
+            fields.add(oneField);
+        }
+
+        return fields;
+    }
+
+    /**
+     * Create a query string from a List<OneField> for Ignite database REST API containing INSERT query
+     */
+    @Override
+    public OneRow setFields(List<OneField> record) throws Exception {
+        StringBuilder sb = new StringBuilder();
+        String fieldDivisor = "";
+        
+        sb.append("(");
+        for (OneField oneField : record) {
+            sb.append(fieldDivisor);
+            fieldDivisor = ", ";
+            switch (DataType.get(oneField.type)) {
+                case INTEGER:
+                case FLOAT8:
+                case REAL:
+                case BIGINT:
+                case SMALLINT:
+                case BOOLEAN:
+                    sb.append(String.valueOf(oneField.val));
+                    break;
+                case VARCHAR:
+                case BPCHAR:
+                case TEXT:
+                case NUMERIC:
+                    sb.append("'" + String.valueOf(oneField.val) + "'");
+                    break;
+                case BYTEA:
+                    sb.append("'" + Hex.encodeHexString((byte[])(oneField.val)) + "'");

```

Comment:
usually getFields and setFields use the same logic (the idea is that if you write and then read the same data, you will get the same data back). Just wondering why use here Hex and not Base64 serialization?

Reply:
That is how Ignite handles such fields :smile:

I have not found the Ignite documentation for this case. However, when Base64 field is sent to Ignite, it handles it some wrong way (it does not return an error, too).